polkadot/xcm/xcm-executor/src/lib.rs

```diff
@@ -508,7 +508,10 @@ impl<Config: config::Config> XcmExecutor<Config> {
 			let assets_taken_from_holding_to_pay_delivery_fees = self
 				.holding
 				.try_take(asset_to_pay_for_fees.clone().into())
-				.map_err(|_| XcmError::NotHoldingFees)?;
+				.map_err(|e| {
+					log::error!(target: "xcm::xcm_executor::take_fee", "Failed to take fees {:?} from holding. {:?}", fee, e);
```

Comment:
@ayevbeosa I woudl better log here `asset_to_pay_for_fees` instead of `fee`, or even better log both
```suggestion
					log::error!(target: "xcm::fees", "Failed to take asset_to_pay_for_fees: {asset_to_pay_for_fees:?} (based on fee: {fee:?}) from holding with error: {e:?}");
```